pennylane/optimize/gradient_descent.py

```diff
@@ -47,86 +48,116 @@ def update_stepsize(self, stepsize):
 
         Args:
             objective_fn (function): the objective function for optimization
-            x (array): NumPy array containing the current values of the variables to be updated
+            *args : Variable length argument list for objective function
             grad_fn (function): Optional gradient function of the
                 objective function with respect to the variables ``x``.
                 If ``None``, the gradient function is computed automatically.
+                Must match shape of autograd derivative.
+            **kwargs : Variable length of keywords for the objective function
 
         Returns:
-            tuple: the new variable values :math:`x^{(t+1)}` and the objective function output
+            tuple(list, float): the new variable values :math:`x^{(t+1)}` and the objective function output
                 prior to the step
         """
 
-        g, forward = self.compute_grad(objective_fn, x, grad_fn=grad_fn)
-        x_out = self.apply_grad(g, x)
+        g, forward = self.compute_grad(objective_fn, args, kwargs, grad_fn=grad_fn)
+        new_args = self.apply_grad(g, args)
 
         if forward is None:
-            forward = objective_fn(x)
+            forward = objective_fn(*args, **kwargs)
 
-        return x_out, forward
+        # unwrap from list if one argument, cleaner return
+        if len(new_args) == 1:
+            return new_args[0], forward
+        return new_args, forward
 
-    def step(self, objective_fn, x, grad_fn=None):
+    def step(self, objective_fn, *args, grad_fn=None, **kwargs):
         """Update x with one step of the optimizer.
 
         Args:
             objective_fn (function): the objective function for optimization
-            x (array): NumPy array containing the current values of the variables to be updated
+            *args : Variable length argument list for objective function
             grad_fn (function): Optional gradient function of the
                 objective function with respect to the variables ``x``.
                 If ``None``, the gradient function is computed automatically.
+                Must match shape of autograd derivative.
+            **kwargs : Variable length of keywords for the objective function
 
         Returns:
-            array: the new variable values :math:`x^{(t+1)}`
+            list: the new variable values :math:`x^{(t+1)}`
         """
-        g, _ = self.compute_grad(objective_fn, x, grad_fn=grad_fn)
-        x_out = self.apply_grad(g, x)
 
-        return x_out
+        g, _ = self.compute_grad(objective_fn, args, kwargs, grad_fn=grad_fn)
+        new_args = self.apply_grad(g, args)
+
+        # unwrap from list if one argument, cleaner return
+        if len(new_args) == 1:
+            return new_args[0]
+
+        return new_args
 
     @staticmethod
-    def compute_grad(objective_fn, x, grad_fn=None):
+    def compute_grad(objective_fn, args, kwargs, grad_fn=None):
```

Comment:
I'm okay with keeping them as `args` and `kwargs`. In a lot of other _internal_ functions across PL we do the same thing (since unpacking in the signature adds additional complexity to the logic required, and is really only needed from a UI standpoint in the user facing method)

Reply:
That's what I was thinking.  When passing internally, I don't want to have to unpack and pack back up again.

Reply:
Fair enough. 🙂 That's for the clarification!